erpnext/selling/doctype/product_bundle/product_bundle.py

```diff
@@ -18,6 +20,14 @@ def validate(self):
 		from erpnext.utilities.transaction_base import validate_uom_is_integer
 		validate_uom_is_integer(self, "uom", "qty")
 
+	def on_trash(self):
+		invoices = frappe.db.get_all("Sales Invoice Item", {"item_code": self.new_item_code, "docstatus": ["!=", 2]}, ["parent"])
```

Comment:
The product bundle can be linked with any transactions, stock entry, purchase invoice. 

Check the `check_if_linked_document_exists` method from `item.py`